src/components/permissions/loginHelpers.ts

```diff
@@ -4,38 +4,37 @@ import { verify, sign } from 'jsonwebtoken'
 import { promisify } from 'util'
 import { PermissionRow, TemplatePermissions, UserInfo } from './types'
 import { compileJWT } from './rowLevelPolicyHelpers'
+import { Organisation } from '../../generated/graphql'
 
 const verifyPromise: any = promisify(verify)
 const signPromise: any = promisify(sign)
 
-const getUsername = async (jwtToken: string) => {
-  let username = 'nonRegistered'
-  if (jwtToken) {
-    try {
-      username = (await verifyPromise(jwtToken, config.jwtSecret)).username
-    } catch (e) {
-      console.log('cannot verify JWT in authorisation header')
-    }
+const getTokenData = async (jwtToken: string) => {
+  try {
+    const data = await verifyPromise(jwtToken, config.jwtSecret)
```

Comment:
previous implementation didn't actual throw an error, but considered token as invalid (and returned a token for `nonRegistered` user. In this case a 500 will be returned (with no error message). Was that intended ?

Reply:
Yeah, kind of. As we discussed in person, I think the nonRegisterd JWT should only be provided when the front-end specifically requests it, and we already have the login in place for that to happen when it's needed.

Reply:
Ok, I still think this is not an internal error 500 material, but maybe 401 (which front end can handle)

Reply:
Sure, the actual try/catch here is basically the same as you had it originally:
```
    try {
      username = (await verifyPromise(jwtToken, config.jwtSecret)).username
    } catch (e) {
      console.log('cannot verify JWT in authorisation header')
    }
```
Happy to change it to something else. What should the return content be on error?

Reply:
The way you've done it i the lates changes looks good (it still returns values which is dealt with further down the track, rather then throwing an error. It's good in this case since the error is easily handled)